portal-impl/src/com/liferay/portal/spring/hibernate/DialectDetector.java

```diff
@@ -102,7 +102,9 @@ else if (dbName.equals("ASE")) {
 			else if (dbName.startsWith("DB2") && (dbMajorVersion >= 9)) {
 				dialect = new DB2Dialect();
 			}
-			else if (StringUtil.startsWith(dbInfo.getDriverName(), "mariadb")) {
+			else if (StringUtil.startsWith(dbInfo.getDriverName(), "mariadb") &&
```

Comment:
@ZoltanTakacs Both the driver name and jdbc url are used to detect mariadb, I am wondering if it is possible to only use one of them or use a different property from databaseMetaData?

Reply:
two thoughts on this: 
I don't know if the product name is supported by all drivers, but maybe that would give us some indication besides the url. https://docs.oracle.com/javase/7/docs/api/java/sql/DatabaseMetaData.html#getDatabaseProductName()

it seems like we are trying to fall through to

```
dialect = DialectFactory.buildDialect(
						new Properties(), connection);
```

does this work for detecting mariaDB? then we could just remove the else if altogether and rely on the dynamic detection

Reply:
never mind ... we are creating the mariaDB dialect ourselves https://github.com/liferay-core-infra/liferay-portal/blob/f4a910dfd426124eed460a819e41bab669661ab3/portal-impl/src/com/liferay/portal/dao/orm/hibernate/MariaDBDialect.java. its not part of hibernate's detection.

I guess this is being used as a marker class? we're not adding anything besides just extending the MysqlDialect

Reply:
@arbetts I see... 
Could you please :

Reply:
@arbetts Could you please modify the checking logic in DBPartionUtil to make it also accept mariaDB? Let's see if it works or not, as you said, there seems no difference between the two.

Tina.

Reply:
so just to be clear you want to alter the usage in https://github.com/liferay-core-infra/liferay-portal/blob/f4a910dfd426124eed460a819e41bab669661ab3/portal-impl/src/com/liferay/portal/db/partition/DBPartitionUtil.java#L228-L234

so that it wraps the datasource for both DBType.MYSQL and DBType.MARIADB?

I'm not sure what should fundamentally determine the dialect; or if its a larger problem to use a driver that supports multiple database servers.

it seems like the jdbc url is accurate to determine the underlying protocol because its always `jdbc:dbprotocol`

Reply:
Yeah, so modifying the `DBPartitionUtil` to accept MariaDB is just hiding the real problem with the dialect detection. We were not using MariaDB server but MySQL. It's an edge case because MariaDB driver also can handle connections to MySQL server too.
- We can get rid of the first condition `StringUtil.startsWith(dbInfo.getDriverName(), "mariadb")` and just simply relying on the JDBC URL. I'll update the PR accordingly.